lib.php

```diff
@@ -343,6 +345,30 @@ function hvp_pluginfile($course, $cm, $context, $filearea, $args, $forcedownload
 
             $itemid = 0;
             break;
+
+        case 'cssfile':
+            if ($context->contextlevel != CONTEXT_SYSTEM) {
+                return false; // Invalid context.
+            }
+
+            $filename = array_pop($args);
+            $filepath = (!$args ? '/' : '/' .implode('/', $args) . '/');
+            $path = $CFG->dirroot . '/mod/hvp' . $filepath . $filename;
+            $cssfile = realpath($path);
+
+            $lifetime = 60 * 60 * 24 * 90;
+
+            header('Etag: "'. get_config('mod_hvp', 'version') .'"');
```

Comment:
Have a look at lib/csslib.php it may refactor out most of this code

Reply:
lib/csslib.php relies on configonly.php, which is only loaded if ABORT_AFTER_CONFIG is set. I tried using this, but got a 'function not found' error.

Reply:
This works just fine:
```
<?php

require_once('config.php');
require_once('lib/configonlylib.php');
require_once('lib/csslib.php');

css_send_cached_css('css.php', 'crap');

```

Reply:
Thanks.